learning.py

```diff
@@ -239,8 +238,8 @@ def PluralityLearner(dataset):
     in the training data.  Makes a baseline for comparison."""
     most_popular = mode([e[dataset.target] for e in dataset.examples])
 
-    def predict(example):
-        "Always return same result: the most popular from the training set."
+    def predict(example=[]):
```

Comment:
It is wrong to make example be an optional argument. The contradict for predict is that it works on an example.